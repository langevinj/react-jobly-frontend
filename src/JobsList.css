.buttons{
    margin-bottom: 1rem;
}

.buttons > button {
    margin: 1rem;
}