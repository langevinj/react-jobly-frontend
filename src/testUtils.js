import React from 'react'
import UserContext from './UserContext'

const testUser = { username: "testuser", firstName: "test", lastName: "testy"}

const UserProvider = ({ children , currUser = testUser }) => (
    <UserContext.Provider value={{currUser}}>
        {children}
    </UserContext.Provider>
);

export { UserProvider }