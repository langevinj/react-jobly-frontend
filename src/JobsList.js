import React, { useState, useEffect } from 'react'
import JoblyApi from './JoblyApi'
import Search from './Search'
import CardList from './CardList'
import { paginateData } from './helpers'
import './JobsList.css'
import PageButtons from './PageButtons'

function JobsList() {
    const [jobs, setJobs] = useState([])

    //used for pagination
    const [pageNum, setPageNum] = useState(0)
    const [pages, setPages] = useState([])

    //set the list of jobs upon rendering
    useEffect(function() {
        filterJobs();
    }, []);

    useEffect(() => {
        function paginatePages(){
            setPages(pages => (paginateData(jobs)))
        }
        paginatePages()
    }, [jobs])

    //filter jobs if search bar is used
    const filterJobs = async (searchTerm) => {
        let res;
        //if the searchterm isn't blank then filter by the term
        if (searchTerm !== "") {
            res = await JoblyApi.filterJobs(searchTerm)
        } else {
            res = await JoblyApi.getJobs();
        }
        setJobs(res)
    }

    //apply to job and change the message for it in the job list
    async function apply(idx){
        let jobId = jobs[idx].id
        let message = await JoblyApi.applyToJob(jobId);
        setJobs(j => j.map(job=>
            job.id === jobId ? { ...job, state: message} : job
        ))
    }

    const list = 
    <>
        <CardList title='jobs' items={pages} toggleJob={toggleJob} pageNum={pageNum} />
        <PageButtons setPageNum={setPageNum} numPages={pages.length} pageNum={pageNum}/>
    </>

    return (
        <div className="JobsList">
            <h4>Applying for jobs is as simple as clicking a button!</h4>
            <Search filter={filterJobs} />
            {!pages[0] ? <div className="mt-3"><p>....no results right now</p></div> : list}
        </div>
    )
}

export default JobsList;