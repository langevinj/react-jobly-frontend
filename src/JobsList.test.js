import React from 'react'
import { render } from '@testing-library/react'
import JobsList from './JobsList'
import { UserProvider } from './testUtils'
import { MemoryRouter } from 'react-router-dom'

it("renders without crashing", function() {
    render(
        <MemoryRouter>
            <UserProvider>
                <JobsList />
            </UserProvider>
        </MemoryRouter>
    )
});

it("matches snapshot", function() {
    const { asFragment } = render(
        <MemoryRouter>
        <UserProvider>
            <JobsList />
        </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
})