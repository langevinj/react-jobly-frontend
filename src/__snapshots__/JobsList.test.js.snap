// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="JobsList"
  >
    <h4>
      Applying for jobs is as simple as clicking a button!
    </h4>
    <form
      class="Search-bar mt-5 form-container"
    >
      <div
        class="input-group"
      >
        <input
          aria-describedby="button-addon1"
          class="form-control"
          id="term"
          name="term"
          placeholder="Search..."
          type="text"
          value=""
        />
        <button
          class="btn-primary rounded"
          id="button-addon1"
        >
          Search
        </button>
      </div>
    </form>
    <div
      class="mt-3"
    >
      <p>
        ....no results right now
      </p>
    </div>
  </div>
</DocumentFragment>
`;
